core/src/Kokkos_Macros.hpp

```diff
@@ -130,45 +131,39 @@
 #if defined(__INTEL_COMPILER)
 #define KOKKOS_COMPILER_INTEL __INTEL_COMPILER
 #elif defined(__INTEL_LLVM_COMPILER)
-#define KOKKOS_COMPILER_INTEL __INTEL_LLVM_COMPILER
+#define KOKKOS_COMPILER_INTEL_LLVM __INTEL_LLVM_COMPILER
 #elif defined(__ICC)
 // Old define
 #define KOKKOS_COMPILER_INTEL __ICC
 #elif defined(__ECC)
 // Very old define
 #define KOKKOS_COMPILER_INTEL __ECC
-#endif
 
+#elif defined(_CRAYC)
 // CRAY compiler for host code
-#if defined(_CRAYC)
 #define KOKKOS_COMPILER_CRAYC _CRAYC
-#endif
 
-#if defined(__APPLE_CC__)
+#elif defined(__APPLE_CC__)
```

Comment:
Since you're now making these definitions strictly mutually exclusive, please check that any places that were testing `KOKKOS_COMPILER_CLANG` are suitably adapted to check `defined(KOKKOS_COMPILER_CLANG) || defined(KOKKOS_COMPILER_APPLECC)`

Reply:
I did now. We only really check for `Clang` when `CUDA` or `OpenMPTarget` is enabled and `AppleClang`  wouldn't work in these cases anyway. There is of course also the check for builtins that need some love anyway.